FoodBookApp/UI/Views/CreateReview/CreateReview2View.swift

```diff
@@ -98,31 +100,60 @@ struct CreateReview2View: View {
                         }
                     }.padding().background(customGray2).cornerRadius(10)
                 }.padding(.horizontal).padding(.top)
-                .navigationTitle("Review")
-                .toolbar{
-                    Button(action: {
-                        
-                        if cleanliness == 0 || waitingTime == 0 || service == 0 || foodQuality == 0 {
-                            showAlert.toggle()
-                        }
-                        else {
-                            // TODO: Step 1: Upload review
-                            print("Uploads this review!")
-                            // Very Nice To Have, but that the "Done" turns into the loading indicator while the review is uploaded, idk how complex it could be though.
+                    .navigationTitle("Review")
+                    .toolbar{
+                        Button(action: {
                             
-                            // Step 1.5: Trigger notification
-                            notify.sendLastReviewNotification(date: Date())
-                                                        
-                            // Step 2: Close sheet
-                            isNewReviewSheetPresented.toggle()
-                        }
-  
-                    }, label: {
-                        Text("Done")
-                            .frame(maxWidth: .infinity)
-                            .padding(.vertical)
-                    })
-                }
+                            if cleanliness == 0 || waitingTime == 0 || service == 0 || foodQuality == 0 {
+                                showAlert.toggle()
+                            }
+                            else {
+                                // Step 1: Upload review
+                                let reviewDate = Date()
+
+                                Task {
+                                    do {
+                                        let reviewImage = try await model.uploadPhoto(image: selectedImage)
+                                        let newReview = Review(content: reviewBody, // FIXME: trim
+                                                               date: reviewDate,
+                                                               imageUrl: reviewImage, // FIXME: handle photo
+                                                               ratings: ReviewStats(
+                                                                cleanliness: cleanliness,
+                                                                foodQuality: foodQuality,
+                                                                service: service,
+                                                                waitTime: waitingTime),
+                                                               selectedCategories: categories,
+                                                               title: reviewTitle, // FIXME: trim
```

Comment:
These trims for some reason collapse the build, something about the compiler being unable to type-check. I believe they could be an issue later, not urgent.